src/vstest.console/Internal/MSBuildLogger.cs

```diff
@@ -197,49 +222,157 @@ private static bool TryGetStackFrameLocation(string stackFrame, out string? line
-            return null;
+            stringBuilder.AppendLine(testResultPrefix + CommandLineResources.StdOutMessagesBanner);
+            AddFormattedOutput(stdOutMessagesCollection, stringBuilder);
         }
 
-        // Remove any occurrence of message splitter.
-        return text.Replace("++++", "____");
+        var stdErrMessagesCollection = GetTestMessages(result.Messages, TestResultMessage.StandardErrorCategory);
+        if (stdErrMessagesCollection.Count > 0)
+        {
+            stringBuilder.AppendLine(testResultPrefix + CommandLineResources.StdErrMessagesBanner);
+            AddFormattedOutput(stdErrMessagesCollection, stringBuilder);
+
+        }
+
+        var dbgTrcMessagesCollection = GetTestMessages(result.Messages, TestResultMessage.DebugTraceCategory);
+        if (dbgTrcMessagesCollection.Count > 0)
+        {
+            stringBuilder.AppendLine(testResultPrefix + CommandLineResources.DbgTrcMessagesBanner);
+            AddFormattedOutput(dbgTrcMessagesCollection, stringBuilder);
+        }
+
+        var addnlInfoMessagesCollection = GetTestMessages(result.Messages, TestResultMessage.AdditionalInfoCategory);
+        if (addnlInfoMessagesCollection.Count > 0)
+        {
+            stringBuilder.AppendLine(testResultPrefix + CommandLineResources.AddnlInfoMessagesBanner);
+            AddFormattedOutput(addnlInfoMessagesCollection, stringBuilder);
+        }
+
+        return stringBuilder.ToString();
     }
 
-    private static string? ReplaceTildaSeparator(string? text)
+    private static void AddFormattedOutput(Collection<TestResultMessage> testMessageCollection, StringBuilder stringBuilder)
     {
-        if (text == null)
+        string testMessageFormattingPrefix = " ";
+        if (testMessageCollection == null)
         {
-            return null;
+            return;
         }
 
-        // Remove any occurrence of message splitter.
-        text = text.Replace("~~~~", "____");
-        // Clean up any occurrence of newline splitter.
-        text = text.Replace("!!!!", "____");
-        // Replace newlines with newline splitter.
-        text = text.Replace(Environment.NewLine, "!!!!");
+        foreach (var message in testMessageCollection)
+        {
+            var prefix = string.Format(CultureInfo.CurrentCulture, "{0}{1}", Environment.NewLine, testMessageFormattingPrefix);
+            var messageText = message.Text?.Replace(Environment.NewLine, prefix).TrimEnd(testMessageFormattingPrefix.ToCharArray());
 
-        return text;
+            if (!messageText.IsNullOrWhiteSpace())
+            {
+                stringBuilder.AppendFormat(CultureInfo.CurrentCulture, "{0}{1}", testMessageFormattingPrefix, messageText);
+            }
+        }
     }
 
+    /// <summary>
+    /// Converts the time span format to readable string.
+    /// </summary>
+    /// <param name="duration"></param>
+    /// <returns></returns>
+    internal static string? GetFormattedDurationString(TimeSpan duration)
```

Comment:
Is it normal that the method is duplicated?

Reply:
Yup. Not ideal, but I wanted to change the format of the timestamp if needed without breaking console logger.